### PR TITLE
Test in both C and C++

### LINKED ISSUE #57: Doesn't build in C (non-C++) mode
It currently doesn't build in C mode, because `NLerp` uses `HMM_Normalize()` instead of `HMM_NormalizeQuaternion()`.

I also doubt that a compiler enforcing real C99 mode would be very happy about your usage of inline in the implementation.  
Having function declarations like `extern float HMM_CosF(float Angle);` and then the implementation as `inline float HMM_CosF(float Angle)` looks very wrong.  
And C99 and newer has that weird thing where functions declared inline in the header must be implemented twice: inline in the header *and* not-inline in the implementation - which is pretty much the opposite of what you're doing there (to get classic/C++ inline behavior one would just supply a `static inline` implementation in the header).

### PR DESCRIPTION
**Pull request by bvisness**
An attempt to fix #57 forever. I wrote a small, custom unit test setup that meets our needs without extraneous features.

This should build in both C and C++ modes with both clang and gcc/g++. But we shall see if Travis behaves...

### CODE REVIEW

**Comment by bvisness** [comment]
@DanielGibson Well, this is annoying. I'm explicitly using `-std=c99` but for some reason Travis is still not seeing `sinf`, `cosf`, etc., which are supposed be standard in c99. Do you know the "right" way to fix this?

**Comment by DanielGibson** [comment]
looks like it can't find the functions at link time, try adding `-lm` :)

**Comment by DanielGibson** [comment]
Also, not sure what `-isystem` is supposed to do here, but in the GCC manpage it sounds like it needs an additional argument (directory to headers?), but either way it doesn't sound like it should be needed there

**Comment by bvisness** [comment]
`-isystem` was an attempt to mimic the includes that were being used with our old test framework. Suffice to say, it didn't work and I can get rid of it.

**Comment by bvisness** [comment]
@DanielGibson Well, clang is happy now but gcc is not. I'm not sure why, since the `-lm` flag should be available for gcc as far as I can tell. Any ideas?

https://travis-ci.org/StrangeZak/Handmade-Math/jobs/218082438

**Comment by DanielGibson** [comment]
Yep. The problem is basically described at http://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc and http://www.network-theory.co.uk/docs/gccintro/gccintro_18.html
If the problem happens depends on whether the compiler defaults to passing --as-needed to the linker or not.

TL;DR: put the -lfoo *after* the source/object file that needs it, in your case make `-lm` the very last argument of the command

**Comment by bvisness** [comment]
There it is! Thanks for the advice on that. I'm still rather new to compiling C projects, as you can see...

@StrangeZak What do you think? Shall I squash and merge?

**Comment by DanielGibson** [comment]
no problem, I'm glad I could help :)

**Comment by StrangeZak** [comment]
Is this good to merged just noticed this sorry. @bvisness
